src/main/java/org/testng/reporters/XMLReporter.java

```diff
@@ -26,6 +26,8 @@
 public class XMLReporter implements IReporter {
 
   public static final String FILE_NAME = "testng-results.xml";
+  private static final String JVM_ARG = "xml.report.name";
```

Comment:
Good idea, and maybe `report` before `xml`: `testng.report.xml.name` (or maybe just `testng.report.name`.